### DISCUSSION TITLE
Reqnroll Visual Studio 2022 Extension v2024.3.152 Released

### DISCUSSION THREAD

**Discussion by gasparnagy** [reqnroll/Reqnroll#259] [Announcements]
_UPDATE (9/18/2024): The extension had compatibility issues with Visual Studio 17.11.4 (maybe also other versions), so we issued a hotfix. The hotfix version is **v2024.4.154**. Please make sure that you upgrade from v2024.3.152 to v2024.4.154 if you experience problems. We track the issue at https://github.com/reqnroll/Reqnroll.VisualStudio/issues/37. Thx for everyone who helped/helps with this issue._

The new version will be automatically installed by default within 1-2 days, otherwise can be downloaded from the [Marketplace](https://marketplace.visualstudio.com/items?itemName=Reqnroll.ReqnrollForVisualStudio2022). Please take a minute to rate the extension there.

## Improvements:

* Find Unused Step Definitions Command: Improved handling of Step Definitions decorated with the 'StepDefinition' attribute. If a Step Definition is used in any Given/Then/When step in a Feature file, the step will no longer show in the 'Find Unused Step Definitions' context menu. (https://github.com/reqnroll/Reqnroll.VisualStudio/issues/24)
* Detect existence of SpecFlow for Visual Studio extension and show a warning (https://github.com/reqnroll/Reqnroll.VisualStudio/issues/16)
* Extend searched config file locations to match the locations searched for the Reqnroll runtime (https://github.com/reqnroll/Reqnroll.VisualStudio/issues/31)
* Offer language updates of Gherkin v29 in the editor (updated from Gherkin v22) (https://github.com/reqnroll/Reqnroll.VisualStudio/issues/33)

## Bug fixes:

* Fix: Using the extension side-by-side with the SpecFlow for Visual Studio extension causes CompositionFailedException (https://github.com/reqnroll/Reqnroll.VisualStudio/issues/25)
* Fix: Define Steps produces incorrect regex for step definition (https://github.com/reqnroll/Reqnroll.VisualStudio/issues/28)

*Contributors of this release (in alphabetical order):* @clrudolphi, @gasparnagy, @jdb0123, @UL-ChrisGlew

**Comment by orifn1** [comment]
looks like it automatically updated(maybe with Visual Studio update) and now it does not work: syntax is not highlighted and there are no reqnroll menu in context menu
![image](https://github.com/user-attachments/assets/9527a3a0-ec56-4f69-9eb3-6e8ef31cd49f)
![image](https://github.com/user-attachments/assets/e1601507-61bc-4d00-808a-bdde29dd54e9)

**Reply by UL-ChrisGlew** [reply]
I'm not able to replicate this, there's been another report raised for similar behaviour but from my end, everything is working okay on the new version:

![image](https://github.com/user-attachments/assets/2fa70af1-a820-45b6-b145-48b6d78a48f7)

**Reply by UL-ChrisGlew** [reply]
@orifn1 - Out of curiosity (no idea if it'll help!) what do you see in the Output tab for Reqnroll?

![image](https://github.com/user-attachments/assets/a002be7a-9626-4929-ac4e-e7d06f684270)

**Reply by anhtuan266** [reply]
I'm facing the same issue. Here is my ActivityLog.xml
<img width="1247" alt="image" src="https://github.com/user-attachments/assets/2a9cdd4d-be36-4a57-bc0c-e08cfa76d10b">

**Reply by orifn1** [reply]
I do not have such option - looks like my vs does not see this extension at all((
![image](https://github.com/user-attachments/assets/23ae70a9-711f-43e4-8fe4-f9210dab6ae3)

**Reply by UL-ChrisGlew** [reply]
What versions of Visual Studio are people using? I'm not on the latest version (17.11.2), so that could be why?

**Reply by UL-ChrisGlew** [reply]
Managed to replicate this on latest version of Visual Studio 2022 (17.11.4)

**Reply by UL-ChrisGlew** [reply]
Debugging the extension, I get the following error thrown:

![image](https://github.com/user-attachments/assets/f0d6f41c-6210-423a-bede-660f146b2e0a)

**Reply by UL-ChrisGlew** [reply]
I have raised an issue tracking this here, if anyone has any additional information please add it there:

https://github.com/reqnroll/Reqnroll.VisualStudio/issues/37

**Comment by orifn1** [comment]
where can I download the previous version of the extension?

**Reply by UL-ChrisGlew** [reply]
Try this?

https://marketplace.visualstudio.com/_apis/public/gallery/publishers/Reqnroll/vsextensions/ReqnrollForVisualStudio2022/2024.2.93/vspackage

You'll need to:

- Uninstall the current Reqnroll for Visual Studio 2022 extension
- Disable automatic updates (for now):
![image](https://github.com/user-attachments/assets/506e6b78-f7c8-4205-82c8-fa365d1c7908)
- Then close Visual Studio 2022 and install the VSIX file downloaded in the URL above.

**Reply by orifn1** [reply]
it helps, thanks

**Comment by JimCooperFromGmailAccount** [comment]
Doesn't work for me on either of the two latest VS 2022 updates.
The *.feature.cs files are created, but nothing in the editor works (no syntax highlighting, no context menu, etc)
The tests themselves run

**Reply by UL-ChrisGlew** [reply]
Hi, we are tracking this here:

https://github.com/reqnroll/Reqnroll.VisualStudio/issues/37

**Comment by gasparnagy** [comment]
The hotfix is on the way, please help validating whether the hotfix solves the problem based on my comment: https://github.com/reqnroll/Reqnroll.VisualStudio/issues/37#issuecomment-2358267727

**Comment by gasparnagy** [comment]
The hotfix is published to the marketplace as **v2024.4.154**.
